pkg/controller/controller_binding.go

```diff
@@ -119,13 +122,16 @@ func (c *controller) reconcileBinding(binding *v1alpha1.Binding) error {
 			binding.Spec.InstanceRef.Name,
 		)
 		glog.Info(s)
-		c.updateBindingCondition(
+		if err := c.updateBindingCondition(
 			binding,
 			v1alpha1.BindingConditionReady,
 			v1alpha1.ConditionFalse,
 			errorWithOngoingAsyncOperation,
 			errorWithOngoingAsyncOperationMessage,
-		)
+		); err != nil {
+			glog.Errorf("updating binding condition (%s)", err)
+			return err
+		}
 		c.recorder.Event(binding, api.EventTypeWarning, errorWithOngoingAsyncOperation, s)
```

Comment:
> What causes updating the condition in status to fail?

Broken pipe, API server goes down, etc

> What happens after? Do we try to reconcile again? Eventually it succeeds?

We periodically re-reconcile everything, yeah.  Eventually it succeeds.